sdk/core/azure-core/src/http/winhttp/win_http_transport.cpp

```diff
@@ -407,6 +419,8 @@ int64_t WinHttpTransport::GetContentLength(
   // Get the content length as a number.
   if (requestMethod != HttpMethod::Head && responseStatusCode != HttpStatusCode::NoContent)
   {
+    handleManager->m_context.ThrowIfCancelled();
```

Comment:
I am not sure about this one.
I feel like queryHeaders will check the headers from memory from the response already read from network, so, it's not I/O.  I would remove this check